Const-cast Coll in AbstractElement.

This is how the implementation of the declaration works, and now clang
is happy.

https://github.com/xmms2/xmms2-devel/commit/2af8dcb1dadee505ee38ef4a9b5a373a7df14f18
--- src/include/xmmsclient/xmmsclient++/coll.h.orig	2011-10-20 14:26:08.000000000 -0500
+++ src/include/xmmsclient/xmmsclient++/coll.h	2022-01-11 21:59:34.000000000 -0600
@@ -612,7 +612,7 @@
 
 		template< typename keyT, typename valT >
 		AbstractElement< keyT, valT >::AbstractElement( const Coll& coll, keyT index )
-			: coll_ (coll), index_( index )
+			: coll_ ( const_cast< Coll& >( coll ) ), index_( index )
 		{
 			xmmsv_coll_ref( coll_.getColl() );
 		}
